src/frontends/onnx/frontend/src/op/average_pool.cpp

```diff
@@ -14,8 +14,31 @@ namespace set_1 {
 ov::OutputVector average_pool(const ov::frontend::onnx::Node& node) {
     return pooling::PoolingFactory(node).make_avg_pool();
 }
-
 }  // namespace set_1
+
+namespace set_7 {
+ov::OutputVector average_pool(const ov::frontend::onnx::Node& node) {
+    return pooling::PoolingFactory(node).make_avg_pool();
```

Comment:
As you can see here https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/utils/pooling_factory.cpp we have some "universal" implementation which translates AveragePool using some common approach.
I see some risks there:
1. A lot of non-documented behavior (all versions could do everything as a last one)
2. Unexpected behavior if user passed a non-supported parameter for an older implementation
3. Increases debug costs in case user has unexpected behavior
4. Harder to maintain code because even we don't know which version we are really supporting :)

My suggestion is: add checks which fails if wrong node is passed.

`CHECK_VALID_NODE(node, !node.has_attribute('ceil_mode') && !node.has_attribute('dilations') && !node.has_attribute('auto_pad'), "AveragePool-7 doesn't support requested functionality");`